Food_Tracker_App/app/src/main/java/com/example/foodtracker/ui/mealPlan/AddNewIngredientMPDialog.java

```diff
@@ -0,0 +1,192 @@
+package com.example.foodtracker.ui.mealPlan;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.example.foodtracker.R;
+import com.example.foodtracker.model.IngredientUnit.IngredientUnit;
+import com.example.foodtracker.model.ingredient.Category;
+import com.example.foodtracker.model.ingredient.Ingredient;
+import com.example.foodtracker.model.ingredient.Location;
+import com.example.foodtracker.model.recipe.SimpleIngredient;
+
+import com.example.foodtracker.utils.Collection;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This dialog is for when an ingredient not in the storage is to be added to a meal plan
+ */
+public class AddNewIngredientMPDialog extends DialogFragment {
```

Comment:
As far as I can tell this class is identical to the `AddRecipeIngredientDialog` class

Reply:
Yes. This is the same as that one, but i don't know how to send the ingredient to the meal plan main screen if i use that one

Reply:
solved